File: easy/python/1_twosum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = {}
        for ind, val in enumerate(nums):
            r = target - val
            if r in d: return [d[r], ind]
            d[val] = ind

File: easy/cpp/1_twosum.cpp
#include <vector>;
#include <unordered_map>;
using namespace std;

class Solution {
    public:
        vector<int> twoSum(vector<int>& nums, int target) {
            unordered_map<int, int> numMap;
            vector<int> ret;
            for(int i = 0; i < nums.size(); i++)
            {
                int num = nums[i];
                if(numMap.count(target-num))
                {
                    ret.push_back(i);
                    ret.push_back(numMap[target-num]);
                }
                else
                {
                    numMap.emplace(nums[i], i);
                }
            }
            return ret;
        }
};